### PR TITLE
Select a class name when opening a builder

### LINKED ISSUE #223: Select a class name when opening a builder
### Feature you want to implement
As I said in snu-quiqcl/artiq-proxy#102, we need to support to designate a specific class name.

For this, I will add a `QInputDialog` to select a class name just after the open button in explorer app is cliecked.

### PR DESCRIPTION
**Pull request by BECATRUE**
This closes #223.

I updated the followings:
- `thread.py`: `ExperimentInfoThread` to emit with all experiments class info of the given experiment path.
- `apps/explorer.py`: After the experiment info is fetched, show a `QInputDialog`.
- `apps/builder.py`: Similar to `explorer.py`.
- Unittests for explorer and builder app.

Now, once we clicked the open button in explorer app, a `QInputDialog` appears.

image

### CODE REVIEW

File: iquip/apps/builder.py [hunk 1]
```diff
@@ -377,11 +380,11 @@ def run(self):
 
         Whenever the experiment is submitted well regardless of whether it runs successfully or not,
         the server returns the run identifier.
-        After submitted, the submitted signal is emitted.
```
Comment (kangz12345) [inline iquip/apps/builder.py:380]: Why is this line removed?

Comment (BECATRUE) [reply iquip/apps/builder.py:380]: I think the description of `submit` signal in the class docstring is enough.
`submitted(rid): The experiment is submitted. The rid is a run identifier.`

In addition, we didn't write signal descriptions in `run()` in dataviwer like [this](https://github.com/snu-quiqcl/iquip/blob/develop/iquip/apps/dataviewer.py#L893).

**Review by kangz12345 — APPROVED** [review]
LGTM
